instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4/common/AbstractNettyChannelPipelineInstrumentation.java

```diff
@@ -153,6 +153,11 @@ public static void removeHandler(
           .getName()
           .startsWith("io.opentelemetry.javaagent.instrumentation.netty.")) {
         pipeline.removeLast();
+      } else if (handler
+          .getClass()
+          .getName()
+          .startsWith("io.opentelemetry.instrumentation.netty.")) {
+        pipeline.removeLast();
```

Comment:
it's not clear to me why this is needed?

Reply:
This was a bit of a thorn in my side. So this instrumentation ensures that when a handler is removed, the associated otel tracing handler is also removed. But in splitting out the netty-4.1 javaagent stuff into a library, the package that the handlers lives in changes from `io.opentelemetry.javaagent.instrumentation.netty.*` to `io.opentelemetry.instrumentation.netty.*`.

Reply:
inside the Java agent, `io.opentelemetry.instrumentation.netty.*` should be shaded to `io.opentelemetry.javaagent.shaded.instrumentation.netty.*`

oh! but this string will get correspondingly shaded. ok, makes sense to me now, thx

Reply:
I didn't know that shading detects and changes string usage.. That makes sense now that I think about it because I tried to use a `matches` with a regex instead of `startsWith` and it wouldn't work.